tests/integration_tests/test_skopt.py

```diff
@@ -0,0 +1,177 @@
+from mock import call
+from mock import patch
+from skopt.space import space
+
+import optuna
+from optuna.testing.sampler import FirstTrialOnlyRandomSampler
+
+
+def test_conversion_from_distribution_to_dimenstion():
+    # type: () -> None
+
+    sampler = optuna.integration.SkoptSampler()
+    study = optuna.create_study(sampler=sampler)
+    with patch('skopt.Optimizer') as mock_object:
+        study.optimize(_objective, n_trials=2, catch=())
+
+        dimensions = [
+            # Original: trial.suggest_uniform('p0', -3.3, 5.2)
+            space.Real(-3.3, 5.2),
+
+            # Original: trial.suggest_uniform('p1', 2.0, 2.0)
+            # => Skipped because `skopt.Optimizer` cannot handle an empty `Real` dimension.
+
+            # Original: trial.suggest_loguniform('p2', 0.0001, 0.3)
+            space.Real(0.0001, 0.3, prior='log-uniform'),
+
+            # Original: trial.suggest_loguniform('p3', 1.1, 1.1)
+            # => Skipped because `skopt.Optimizer` cannot handle an empty `Real` dimension.
+
+            # Original: trial.suggest_int('p4', -100, 8)
+            space.Integer(-100, 8),
+
+            # Original: trial.suggest_int('p5', -20, -20)
+            # => Skipped because `skopt.Optimizer` cannot handle an empty `Real` dimension.
+
+            # Original: trial.suggest_discrete_uniform('p6', 10, 20, 2)
+            space.Integer(0, 5),
+
+            # Original: trial.suggest_discrete_uniform('p7', 0.1, 1.0, 0.1)
+            space.Integer(0, 8),
+
+            # Original: trial.suggest_discrete_uniform('p8', 2.2, 2.2, 0.5)
+            # => Skipped because `skopt.Optimizer` cannot handle an empty `Real` dimension.
+
+            # Original: trial.suggest_categorical('p9', ['9', '3', '0', '8'])
+            space.Categorical(('9', '3', '0', '8'))
+        ]
+        assert mock_object.mock_calls[0] == call(dimensions)
+
+
+def test_suggested_value():
```

Comment:
I think tests for objective functions that returns `NaN` or `None` are missing. Could you add such test cases to confirm `SkoptSampler` handles them correctly?

Reply:
I will add tests for `NaN` and `None`. Thank you for pointing it out.

Reply:
Added: https://github.com/pfnet/optuna/pull/399/commits/86f9f83c90ea195426e9d0e65e0f21df267aed4e

Because the type of objective function doesn't allow that the function returns `None`, I omitted a test case for `None` return value.

Reply:
You're right. Thanks.